### PR TITLE
Cesar/eancode

### PR DESCRIPTION
**Pull request by denis1729**
good work Cesar

### CODE REVIEW

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            } else {
+                num = num * MPLT_IMPAR;
+            }
+            suma = suma + num;
+        }
+        int numcom = MPLT_MOD - (suma % MPLT_MOD);
+        boolean res;
+        if (numcom == numComprobacion) {
+            res = true;
+            System.out.println("ES VALIDO");
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        } else {
+            res = false;
+            System.out.println("ES INVALIDO");
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class EanTest {
+    private Ean val;
+
+    /**
+     * Starting the before annotation.
+     */
+    @Before
+    public void before() {
+        val = new Ean();
+        System.out.println("before");
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        res = val.validar();
+        assertEquals(true, res);
+    }
+
+    /**
+     * test for Validar failed.
+     */
+    @Test
+    public void testValidarFailed() {
+        System.out.println("no pasa");
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        val = new Ean();
+        System.out.println("before");
+    }
+
+    /**
+     * test for Validar pass.
+     */
+    @Test
+    public void testValidarPass() {
+        System.out.println("pasa");
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

**Comment by carledriss** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        digitos = cadena.substring(0, cadena.length() - 1);
+        numComprobacion = Integer.parseInt(cadena.substring(cadena.length() - 1));
+    }
+
+    /**
+     *
+     * @return Returning false or true value
+     */
+
+    public boolean validar() {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: As we talked in the past class 
This method should receive the String ean code

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            if (num % 2 == 0) {
+                num = num * MPLT_PAR;
+            } else {
+                num = num * MPLT_IMPAR;
+            }
+            suma = suma + num;
+        }
+        int numcom = MPLT_MOD - (suma % MPLT_MOD);
+        boolean res;
+        if (numcom == numComprobacion) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: you can return directly this conditional

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * test for Validar pass.
+     */
+    @Test
+    public void testValidarPass() {
+        boolean res;
+        val.setCadena("4003301018398");
+        res = val.validar();
+        assertEquals(true, res);
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: You can use an assertTrue here

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * test for Validar failed.
+     */
+    @Test
+    public void testValidarFailed() {
+        boolean res;
+        val.setCadena("4003301018392");
+        res = val.validar();
+        assertEquals(false, res);
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: You can use an assertFalse here

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        for (String i : aux) {
+            int num = Integer.parseInt(i);
+            if (num % 2 == 0) {
+                num = num * MPLT_PAR;
+            } else {
+                num = num * MPLT_IMPAR;
+            }
+            suma = suma + num;
+        }
+        numcom = MPLT_MOD - (suma % MPLT_MOD);
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: This formula is wrong 
Please review again the problem description

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    public boolean validar(final String cadena) {
+        int numcom;
+        String digitos = cadena.substring(0, cadena.length() - 1);
+        numComprobacion = Integer.parseInt(cadena.substring(cadena.length() - 1));
+        String[] aux = digitos.split("");
+        int suma = 0;
+        for (String i : aux) {
+            int num = Integer.parseInt(i);
+            if (num % 2 == 0) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: You can use here a ternary operator.

File: src/test/java/org/fundacionjala/coding/cesar/EanTest.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
+package org.fundacionjala.coding.cesar;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ *
+ * @author admin-hp
+ */
+public class EanTest {
```
Comment (carledriss) [inline src/test/java/org/fundacionjala/coding/cesar/EanTest.java]: It is missing one test when checksum is Zero

File: src/main/java/org/fundacionjala/coding/cesar/Ean.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        numComprobacion = Integer.parseInt(cadena.substring(cadena.length() - 1));
+        String[] aux = digitos.split("");
+        int suma = 0;
+        for (String i : aux) {
+            int num = Integer.parseInt(i);
+            num = num % 2 == 0 ? num * MPLT_PAR : num * MPLT_IMPAR;
+            suma = suma + num;
+        }
+        numcom =  suma %  MPLT_MOD  ==  0  ?  0  :  MPLT_MOD  - suma %  MPLT_MOD;
+        return numcom == numComprobacion;
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/cesar/Ean.java]: This may be refactored like this.
```java
    boolean validar(final String cadena) {
        String[] numbers = cadena.substring(0, cadena.length() - 1).split("");
        int sum = 0;
        for (String number : numbers) {
            int num = Integer.parseInt(number);
            sum += num % 2 == 0 ? num : num * MPLT_IMPAR;
        }
        int checkSum =  sum %  MPLT_MOD  ==  0  ?  0  :  MPLT_MOD  - sum %  MPLT_MOD;
        int lastDigit = Integer.parseInt(cadena.substring(cadena.length() - 1));
        return checkSum == lastDigit;
    }
```

**Review by carledriss — APPROVED** [review]
LGTM! 👍
